File: Python/SQLAlchemy_ORM/bll/services/customer_service.py
from dal.repositories.customer_repository import CustomerRepository
from dal.data.database import SessionLocal
from dal.entities.customer import Customer

class CustomerService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = CustomerRepository(self.db)

    def get_by_id(self, id: str):
        entity = self.repo.get_by_id(id)
        if not entity:
            raise ValueError(f"Entity {id} not found")
        return entity

    def get_all(self):
        return self.repo.get_all()

    def create(self, entity: Customer) -> Customer:
        return self.repo.create(entity)

    def update(self, id: str, entity: Customer) -> Customer:
        entity = self.repo.update(id, entity)
        if not entity:
            raise ValueError(f"Entity {id} not found for update")
        return entity

    def delete(self, id: int):
        entity = self.repo.delete(id)
        if not entity:
            raise ValueError(f"Entity {id} not found for deletion")
        return entity

File: Node/Sequelize_ORM/src/bll/services/customerService.js
const CustomerRepository = require('../../dal/repositories/customerRepository.js');

class CustomerService {
  async getAll() {
    return await CustomerRepository.getAll();
  }

  async getById(id) {
    return await CustomerRepository.getById(id);
  }

  async create(data) {
    return await CustomerRepository.create(data);
  }

  async update(id, data) {
    return await CustomerRepository.update(id, data);
  }

  async delete(id) {
    return await CustomerRepository.delete(id);
  }
}

module.exports = new CustomerService(); 